assets/js/components/OptIn.js

```diff
@@ -21,119 +21,75 @@
+	const enabled = useSelect( ( select ) => select( CORE_USER ).isTrackingEnabled() );
+	const saving = useSelect( ( select ) => select( CORE_USER ).isSavingUserTracking() );
+	const error = useSelect( ( select ) => select( CORE_USER ).getErrorForAction( 'saveUserTracking', [ ! enabled ] ) );
 
-	async handleOptIn( e ) {
+	const { saveUserTracking } = useDispatch( CORE_USER );
+	const handleOptIn = useCallback( ( e ) => {
 		const checked = !! e.target.checked;
-		const trackingUserOptInKey = getMetaKeyForUserOption( 'googlesitekit_tracking_optin' );
 
 		toggleTracking( checked );
-
 		if ( checked ) {
-			await trackEvent( 'tracking_plugin', this.props.optinAction );
-		}
-
-		try {
-			await apiFetch( {
-				path: '/wp/v2/users/me',
-				method: 'POST',
-				data: {
-					meta: {
-						[ trackingUserOptInKey ]: checked,
-					},
-				},
-			} );
-			this.setState( {
-				optIn: checked,
-				error: false,
-			} );
-		} catch ( err ) {
-			this.setState( {
-				optIn: ! checked,
-				error: {
-					errorCode: err.code,
-					errorMsg: err.message,
-				},
-			} );
+			trackEvent( 'tracking_plugin', optinAction );
 		}
-	}
-
-	render() {
-		const {
-			optIn,
-			error,
-		} = this.state;
 
-		const {
-			id,
-			name,
-			className,
-		} = this.props;
+		saveUserTracking( checked );
+	}, [ enabled, optinAction ] );
 
-		const labelHTML = sprintf(
-			/* translators: %s: privacy policy URL */
-			__( 'Help us improve the Site Kit plugin by allowing tracking of anonymous usage stats. All data are treated in accordance with <a href="%s" target="_blank" rel="noopener noreferrer">Google Privacy Policy</a>', 'google-site-kit' ),
-			'https://policies.google.com/privacy'
-		);
+	if ( enabled === undefined ) {
+		return null;
+	}
 
-		return (
-			<div className={ classnames(
-				'googlesitekit-opt-in',
-				className
-			) }>
-				<Checkbox
-					id={ id }
-					name={ name }
-					value="1"
-					checked={ optIn }
-					onChange={ this.handleOptIn }
-				>
-					<span
-						dangerouslySetInnerHTML={ sanitizeHTML(
-							labelHTML,
-							{
-								ALLOWED_TAGS: [ 'a' ],
-								ALLOWED_ATTR: [ 'href', 'target', 'rel' ],
-							}
-						) }
-					/>
-				</Checkbox>
-				{ error &&
+	const labelHTML = sprintf(
+		/* translators: %s: privacy policy URL */
+		__( 'Help us improve the Site Kit plugin by allowing tracking of anonymous usage stats. All data are treated in accordance with <a href="%s" target="_blank" rel="noopener noreferrer">Google Privacy Policy</a>', 'google-site-kit' ),
+		'https://policies.google.com/privacy'
+	);
+
+	const allowedDOM = {
+		ALLOWED_TAGS: [ 'a' ],
+		ALLOWED_ATTR: [ 'href', 'target', 'rel' ],
+	};
+
+	return (
+		<div className={ classnames( 'googlesitekit-opt-in', className ) }>
+			<Checkbox
+				id={ id }
+				name={ name }
+				value="1"
+				checked={ enabled }
+				disabled={ saving }
+				onChange={ handleOptIn }
+			>
+				<span dangerouslySetInnerHTML={ sanitizeHTML( labelHTML, allowedDOM ) } />
```

Comment:
While we're refactoring this, we should be able to clean this up by using `createInterpolateElement` instead.

Reply:
Updated.